### PR TITLE
bugfix: fix deadlock when connections expire due to full buffers

### PR DESCRIPTION
**Pull request by kegsay**

### CODE REVIEW

File: sync3/connmap.go [hunk 1]
```diff
@@ -115,9 +123,7 @@ func (m *ConnMap) Conn(cid ConnID) *Conn {
 	}
 	// e.g buffer exceeded, close it and remove it from the cache
 	logger.Info().Str("conn", cid.String()).Msg("closing connection due to dead connection (buffer full)")
-	m.mu.Lock()
```
Comment (DMRobertson) [inline sync3/connmap.go:118]: Presumably the point is that this lock would block because the mutex is already locked on -130.

**Review by DMRobertson — APPROVED** [review]
Is there any way we can spot these deadlocks at test time?

**Comment by kegsay** [comment]
I modified a test for this particular case which acts as a regression test.

File: sync3/connmap.go (hunk 1 above)
Comment (kegsay) [reply sync3/connmap.go:118]: Bingo

**Pull request merged**